### ISSUE TITLE
API exception if transaction is deleted by rule

### ISSUE THREAD

**Issue by PlutoHDDev** [firefly-iii/firefly-iii#8665]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

The issue has already been reported in #5177 but it hasn't been fixed. If there is a rule that deletes transactions created by the API or the data importer in certain cases (in my case, if it has the tag pending), an error is triggered (as described in the issue).

200032: Cannot find transaction. Possibly, a rule deleted this transaction after its creation.

This is understandable so far, but in this case the deletion is intended. Therefore, I don't want to receive an email with such an error every time.

### Debug information

Debug information generated at 2024-03-12 08:32:57 for Firefly III version **v6.1.10**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.10 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.3 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1003, base <span>#</span>81</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Berlin + Europe/Berlin</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>de_DE + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2024-03-12 00:00:00 (8 hours ago)</td></tr><tr><td>Mailer</td><td>smtp</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has rules">:wrench:</span></td></tr><tr><td>Session start</td><td></td></tr><tr><td>Session end</td><td></td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>de_DE</td></tr><tr><td>User locale</td><td>de_DE</td></tr><tr><td>Locale(s) supported</td><td>de_DE.utf8: :white_check_mark:<br>de_DE.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:123.0) Gecko/20100101 Firefox/123.0</td></tr></tbody></table>

### Expected behaviour

It would be great if you could set how this case is handled. I want the deletion so that the import only happens when the booking is no longer pending. Therefore, I don't care about the error and I don't need to receive any emails about it.

### Steps to reproduce

1. Create a rule which deletes a transaction on creation
2. Create a transaction via API or data importer which is covered by this rule

**Comment by JC5** [comment]
Sorry, since this could also be an indication of other problems I'm not planning to change this behavior.

**Comment by JC5** [comment]
Yes, this is confusing but it is the final state. The data importer will not complain (that is the setting for), but Firefly III will always complain about it. This is more of a sanity check than anything else. Suddenly disappearing transactions are something to watch for, so I've opted to keep the error in place.

I hope this explains it :)

Cheers,
James
